File: clojure/cat.clj
#!/usr/bin/env clj
;
;  cat.clj
;
;  An implementation of the UNIX cat command in Clojure. Solution borrows from
;  http://stackoverflow.com/questions/2034059/how-to-read-lines-from-stdin-in-in-clojure
;

(doall (map println
            (line-seq (java.io.BufferedReader. *in*))))

File: julia/cat.jl
#!/usr/bin/env julia
#
#  cat.jl
#

if length(ARGS) > 0
    for filename in ARGS
        f = open(filename)
        for l in eachline(f)
            write(STDOUT, l)
        end
        close(f)
    end
else
    for l in eachline(STDIN)
        write(STDOUT, l)
    end
end
